src/API/Google/Settings.php

```diff
@@ -259,7 +258,21 @@ protected function create_shipping_service( string $country, string $method, str
 			// Todo: Set the pickup service (PickupCarrierService).
 		}
 
-		$service->setRateGroups( $this->create_rate_groups( $currency, $rate ) );
+		$rate_groups = [];
+		// Create a rate group for each shipping class (if any).
+		if ( is_array( $options ) && ! empty( $options['shipping_class_rates'] ) ) {
+			foreach ( $options['shipping_class_rates'] as ['class' => $class, 'rate' => $rate] ) {
+				$rate_groups[] = $this->create_rate_group_object(
+					$currency,
+					$rate,
+					$method,
+					[ $class ]
+				);
+			}
+		}
+		// Create a main rate group for the service.
+		$rate_groups[] = $this->create_rate_group_object( $currency, $rate, $method );
```

Comment:
As mentioned in the overview, this `$rate` uses the last value from the foreach loop and not the original rate passed to the function.

Reply:
Fixed in https://github.com/woocommerce/google-listings-and-ads/pull/1175/commits/ff7f90ce04e6254cad98974eba2079c26dd2f8fa